src/Bootstrappers/BatchTenancyBootstrapper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Bus\DatabaseBatchRepository;
+use Illuminate\Support\Facades\DB;
+use ReflectionClass;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class BatchTenancyBootstrapper implements TenancyBootstrapper
+{
+    private $previousConnection = null;
+
+    public function bootstrap(Tenant $tenant)
+    {
+        $batchRepository = app(BatchRepository::class);
+
+        if ($batchRepository instanceof DatabaseBatchRepository) {
+            /**
+             * Access the resolved batch repository instance and update its connection to use the tenant connection
+             */
+            $batchRepositoryReflection = new ReflectionClass($batchRepository);
+            $connectionProperty        = $batchRepositoryReflection->getProperty('connection');
+            $connectionProperty->setAccessible(true);
+            $connection = $connectionProperty->getValue($batchRepository);
+
+            $this->previousConnection = $connection;
+
+            $connectionProperty->setValue($batchRepository, DB::connection('tenant'));
```

Comment:
Aren't there any getters/setters whatsoever?

If not, and there's a valid use case for modifying the connection (like what we do here), then we could PR the getters & setters to laravel/framework. They typically merge simple PRs like that when they come with good explanations of why the changes are useful.

Reply:
@abrardev99 could you look into this? Seems like OP can't finish the PR now.

Reply:
Sure.

Reply:
> Aren't there any getters/setters whatsoever?

I see there are no getters and setters for connection. I'll send the PR to Laravel, and after merging, I'll implement it in this review. Then we also need to use the Latest Laravel release. 

To ensure PR doesn't reject, I'll make the changes in `DatabaseBatchRepository` class. Correct? 
And changes will be. 
```
<?php

namespace Illuminate\Bus;

use Carbon\CarbonImmutable;
use Closure;
use DateTimeInterface;
use Illuminate\Database\Connection;
use Illuminate\Database\PostgresConnection;
use Illuminate\Database\Query\Expression;
use Illuminate\Support\Str;

class DatabaseBatchRepository implements PrunableBatchRepository
{
    /**
     * The batch factory instance.
     *
     * @var \Illuminate\Bus\BatchFactory
     */
    protected $factory;

    /**
     * The database connection instance.
     *
     * @var \Illuminate\Database\Connection
     */
    protected $connection;

    /**
     * The database table to use to store batch information.
     *
     * @var string
     */
    protected $table;

    /**removed existing methods to showcase current changes**/
    
    /**
     * Set the desired connection for the batch.
     *
     * @return \Illuminate\Database\Connection
     */
    public function getConnection()
    {
        return $this->connection;
    }

    /**
     * get the batch current connection
     *
     * @param  \Illuminate\Database\Connection  $connection
     */
    public function setConnection(Connection $connection)
    {
        $this->connection = $connection;
    }
}

```

Then we can make use of these methods in this PR and get rid of reflection methods.

Reply:
The docblocks are switched up and not formatted well but yeah, you can try to PR those methods.

Reply:
Ah, How dumb 😬. Thanks for pointing out.

Reply:
PR got merged. https://github.com/laravel/framework/pull/43869#event-7268635821

I'll update this PR once Laravel releases the latest tag this Tuesday.

Reply:
Updated 👍 and now resolving the review.